postgresql/database.go

```diff
@@ -84,79 +78,242 @@ func debugLog(query string, args []interface{}, err error, start int64, end int6
 	}
 }
 
-func init() {
+// Returns the underlying *sqlx.DB instance.
+func (s *source) Driver() interface{} {
```

Comment:
I agree with that, let's get this branch polished, and then can make an easy decision later on what you want Driver() to return.